app/models/category_with_entity.rb

```diff
@@ -0,0 +1,14 @@
+class CategoryWithEntity < ApplicationRecord
+  belongs_to :category
+  belongs_to :entity
+
+  after_validation :update_total_amount
+
+  def update_total_amount
+    cat = Category.find(category_id)
+    ent = Entity.find(entity_id)
+    cat.update(total_amount: cat.total_amount + ent.amount)
+    # cat.total_amount = category.total_amount + ent.amount
+    # cat.save
```

Comment:
- [ ] Here as well. I would recommend removing the commented code to make the codebase cleaner.